apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/IdHasher.java

```diff
@@ -0,0 +1,102 @@
+package gov.cms.bfd.pipeline.sharedutils;
+
+import com.google.common.annotations.VisibleForTesting;
+import gov.cms.bfd.sharedutils.exceptions.BadCodeMonkeyException;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import org.apache.commons.codec.binary.Hex;
+
+/**
+ * Utility class to encapsulate the hash algorithm and settings used to compute secure hash values
+ * for ID values such as HICN and MBI. Code adopted from RifLoader.
+ */
+public class IdHasher {
+
+  private final Config config;
+  private final SecretKeyFactory secretKeyFactory;
+
+  public IdHasher(Config config) {
+    this.config = config;
+    secretKeyFactory = createSecretKeyFactory();
+  }
+
+  @VisibleForTesting
+  public static IdHasher createInstanceForTestingOnly() {
+    return new IdHasher(new Config(1000, "nottherealpepper".getBytes(StandardCharsets.UTF_8)));
+  }
```

Comment:
Might be better suited just having this in the test classes themselves and not in normal service code?  Could also be a testing package utility if there was a DRY concern.

Reply:
Good observation.  Yeah this one is all about DRY.  Test code isn't shared between maven modules without an explicit extra dependency in the pom, which I find to be painful.  It can be easier and harmless just to add a properly documented static method.

Reply:
It would be my own opinion that a single constructor call is not enough to be considered a DRY concern, I was more just speaking generally.  Personally, I think having testing code in deployed code is less desirable, and typically the DRY concept doesn't really apply to testing like it does to development code, and in fact I believe it's fine for tests to be very redundant and verbose for the sake of isolation.  Just a thought.